### DISCUSSION TITLE
Bug with destroy / unmount component

### DISCUSSION THREAD

**Discussion by pwFoo** [vuejs/petite-vue#172] [Q&A]
Looks like a bug during destroy / unmount component. After `@vue:unmount` is executed I get error messages in console.

<img width="508" alt="image" src="https://user-images.githubusercontent.com/333408/182906840-78b5929f-28cd-4c99-8768-66e09ec25760.png">

Looks like template triggeres a reactive data getter during component is destroyed and getter function no more exists?

**Template**
```
        <template id="template-match">
            <h2>Match {{match._['#']}}</h2>
            <div><span>{{home.name}}</span> <span>{{match.score_home}}</span></div>
            <div><span>{{guest.name}}</span> <span>{{match.score_guest}}</span></div>
        </template>
```

**Component**
```
            match = function() {
                return {
                    $template: '#template-match',
                    get match() {
                        return page.matches.find(m => m._ && page.params && page.params.match && m._['#'] == page.params.match)
                    },
                    get home() {
                        return page.teams.find(t => this.match && this.match.team_home['#'])
                    },
                    get guest() {
                        return page.teams.find(t => this.match && this.match.team_guest['#'])
                    }
                }
            }
```

I'm new with (petite-)vue and code isn't optimized, but shouldn't throw errors that way?

**Comment by PSanilP** [comment]
Same here. Throwing up an error when you re-render the div with new data.
I am wondering how to explicitly unmount

**Comment by PSanilP** [comment]
Have you tried v-bind:key=someuniquekey ?
I am not sure if petitevue has same concept as vue2 https://michaelnthiessen.com/key-changing-technique
